benchmark/src/index.ts

```diff
@@ -0,0 +1,224 @@
+  };
+}
+
+function cachedBuild(): Benchmark {
+  const timings: number[] = [];
+  let total = 0;
+  for (let i = 0; i < REPETITIONS; i++) {
+    const start = new Date().getTime();
+    cp.execSync(`${TURBO_BIN} run build`, DEFAULT_EXEC_OPTS);
+    const end = new Date().getTime();
+    const timing = end - start;
+    total += timing;
+    timings.push(timing);
+  }
+  const avg = total / REPETITIONS;
+  const max = Math.max(...timings);
+  const min = Math.min(...timings);
+  return {
+    name: "Cached Build - no changes",
+    value: avg,
+    unit: "ms",
+    range: String(max - min),
+  };
+}
+
+function saveCache() {
+  // Remove any existing backup
+  if (fs.existsSync(ALT_CACHE_PATH)) {
+    fs.rmSync(ALT_CACHE_PATH, { recursive: true });
+  }
+  // copy the current cache to the backup
+  if (fs.existsSync(DEFAULT_CACHE_PATH)) {
+    fse.copySync(DEFAULT_CACHE_PATH, ALT_CACHE_PATH, { recursive: true });
+  } else {
+    // make an empty cache
+    fs.mkdirSync(ALT_CACHE_PATH, { recursive: true });
+  }
+}
+
+function restoreSavedCache() {
+  // Remove any existing cache
+  if (fs.existsSync(DEFAULT_CACHE_PATH)) {
+    fs.rmSync(DEFAULT_CACHE_PATH, { recursive: true });
+  }
+  // Copy the backed-up cache to the real cache
+  fse.copySync(ALT_CACHE_PATH, DEFAULT_CACHE_PATH, { recursive: true });
+}
+
+function cachedBuildWithDelta(): Benchmark {
+  // Save existing cache just once, we'll restore from it each time
+  saveCache();
+
+  // Edit a file in place
+  const file = path.join(
+    REPO_PATH,
+    "packages",
+    "crew",
+    "important-feature-0",
+    "src",
+    "lib",
+    "important-component-0",
+    "important-component-0.tsx"
+  );
+  const contents = fs.readFileSync(file).toString("utf-8");
+  // make a small edit
+  const updated = contents.replace("-0!", "-0!!");
+  fs.writeFileSync(file, updated);
+
+  const timings: number[] = [];
+  let total = 0;
+  for (let i = 0; i < REPETITIONS; i++) {
+    // Make sure we're starting with the cache from before we make the source code edit
+    restoreSavedCache();
+    const start = new Date().getTime();
+    cp.execSync(`${TURBO_BIN} run build`, DEFAULT_EXEC_OPTS);
+    const end = new Date().getTime();
+    const timing = end - start;
+    total += timing;
+    timings.push(timing);
+  }
+  const avg = total / REPETITIONS;
+  const max = Math.max(...timings);
+  const min = Math.min(...timings);
+  return {
+    name: "Cached Build - source code change",
+    value: avg,
+    unit: "ms",
+    range: String(max - min),
+  };
+}
+
+function cachedBuildWithDependencyChange(): Benchmark {
+  // Save existing cache just once, we'll restore from it each time
+  saveCache();
+
+  // Edit a dependency
+  const file = path.join(REPO_PATH, "apps", "navigation", "package.json");
+  const contents = JSON.parse(fs.readFileSync(file).toString("utf-8"));
+  contents.dependencies["crew-important-feature-0"] = "*";
+  fs.writeFileSync(file, JSON.stringify(contents, null, 2));
```

Comment:
Nit: `Edit` -> `Add` since `crew-important-feature-0` is not originally a dependency of `apps/navigation`